src/library/coverartcache.cpp

```diff
@@ -114,16 +105,27 @@ QPixmap CoverArtCache::requestCover(const CoverInfo& requestInfo,
     QFutureWatcher<FutureResult>* watcher = new QFutureWatcher<FutureResult>(this);
     QFuture<FutureResult> future = QtConcurrent::run(
             this, &CoverArtCache::loadCover, requestInfo, pRequestor,
-            requestReference, desiredWidth, signalWhenDone);
+            desiredWidth, signalWhenDone);
     connect(watcher, SIGNAL(finished()), this, SLOT(coverLoaded()));
     watcher->setFuture(future);
     return QPixmap();
 }
 
+//static
+void CoverArtCache::requestCover(const Track* pTrack,
+                         const QObject* pRequestor) {
+    CoverArtCache* pCache = CoverArtCache::instance();
+    if (pCache == nullptr || pTrack == nullptr) return;
+
+    CoverInfo info = pTrack->getCoverInfo();
+    // trackLocation is still empty here
```

Comment:
I am not sure, if it is possible to have an assertion here. 
I will add better comment and a todo here. 
```
    // trackLocation can be still empty here
    // TODO(DSC) it is an design issue that we have
    // redundant trackLocation info here
```
In the meanwhile I have prepared a branch that solve the issue by better suited CoverInfo classes. Since it is quite big, I will issue it as a follow up commit.